### PR TITLE
Release v0.18.0

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
This PR updates the release actions for better release notes and fixes a few final issues for v0.18.0.

The issues:
- We didn't properly gate the CLI options in wash-lib's registry module, which is now fixed with an internal `mod`
- When the NATS pidfile was empty, we instead just sent the `stop` signal, which would stop even unmanaged NATS servers. Now, we return an error which will be reflected in the `wash down` output.

~I've also begun migrating our https://github.com/wasmCloud/homebrew-wasmcloud and https://github.com/wasmCloud/chocolatey-wash repositories here. I foresee some iteration especially with editing files via `sed` (which I may use templates to make less hacky looking) so in order to test via workflow dispatch and in a separate PR I'm just committing some base actions and the metadata from those repos.~

## Related Issues
~Begins work on #615~

## Release Information
v0.18.0

## Consumer Impact
Refer to the release notes for v0.18.0 for the full consumer impact.

## Testing
Tested individual issues manually to confirm resolution

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

File: .github/workflows/release_choco.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  # Only runs if the regular release fails
+  failure_run:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          echo "Release workflow failed, not releasing to homebrew"
```
Comment (thomastaylor312) [inline .github/workflows/release_choco.yml]: s/homebrew/chocolatey

File: crates/wash-lib/src/registry.rs [hunk 2]
```diff
@@ -29,6 +27,109 @@ const OCI_MEDIA_TYPE: &str = "application/vnd.oci.image.layer.v1.tar";
 // straight up stolen from oci_distribution::Reference
 pub const REFERENCE_REGEXP: &str = r"^((?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?/)?[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][[:xdigit:]]{32,}))?$";
 
+#[cfg(feature = "cli")]
+mod cli {
```
Comment (thomastaylor312) [inline crates/wash-lib/src/registry.rs]: Can you just move this mod to its own file in the CLI dir?

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]

File: .github/workflows/release_choco.yml (hunk 1 above)
Comment (brooksmtownsend) [reply .github/workflows/release_choco.yml]: removed

File: crates/wash-lib/src/registry.rs (hunk 2 above)
Comment (brooksmtownsend) [reply crates/wash-lib/src/registry.rs]: Yessir

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**